web/ui/src/Services/AppsController.js

```diff
@@ -229,40 +180,126 @@
+
+
 
+        /*
+         * PRIVATE FUNCTIONS
+         */
         function refreshTemplates(){
-            resourcesFactory.get_app_templates(false, function(templatesMap) {
-                var templates = [];
-                for (var key in templatesMap) {
-                    var template = templatesMap[key];
-                    template.Id = key;
-                    templates[templates.length] = template;
+            resourcesFactory.get_app_templates(false, function(templates) {
+                $scope.templates.data = utils.mapToArr(templates);
+            });
+        }
+
+        // poll for apps that are being deployed
+        $scope.deployingServices = [];
+        var lastPollResults = 0;
+        function getDeploying(){
+            resourcesFactory.get_active_templates(function(data) {
+                if(data){
+                    $scope.deployingServices = data;
                 }
-                $scope.templates.data = templates;
+
+                //if we have fewer results than last poll, we need to refresh our table
+                //TODO - better checking for deploying apps
+                if(lastPollResults > $scope.deployingServices.length){
+                    servicesFactory.update();
+                }
+                lastPollResults = $scope.deployingServices.length;
             });
         }
 
-        // Get a list of templates
-        refreshTemplates();
+        function removeService(service) {
+            resourcesFactory.remove_service(service.id, function(){
+                // TODO - once the backend updates deleted
+                // services, this should be removed
+                // FIXME - should not modify servicesFactory's
+                // objects!
+                for(var i = 0; i < $scope.apps.length; i++){
+                    // find the removed service and remove it
+                    if($scope.apps[i].id === service.id){
+                        $scope.apps.splice(i, 1);
+                        return;
+                    }
+                }
+            });
+        }
 
-        servicesFactory.activate();
-        // Get a list of deployed apps
-        servicesFactory.update().then(function update(){
-            $scope.services.data = servicesFactory.serviceTree;
+        function deleteTemplate(templateID){
+            resourcesFactory.delete_app_template(templateID, refreshTemplates);
+        }
 
-            // if only isvcs are deployed, and this is the first time
-            // running deploy wizard, show the deploy apps modal
-            if(!$cookies.autoRunWizardHasRun && $scope.services.data.length === 1){
-                $scope.modalAddApp();
+        // init stuff for this controller
+        function init(){
+            // check is location.hostname is an IP
+            var re = /\b(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\b/;
```

Comment:
This same check is ServiceDetailsController.js - both controllers should use the same implementation somehow.
Maybe a new method on resourcesFactory?